### ISSUE TITLE
MISP modules - manual installation person behind keyboard error 

### ISSUE THREAD

**Issue by open-source-rs** [MISP/MISP#5452]
tried using the manual install of modules from this resource:
https://github.com/MISP/misp-modules

some errors during install which led me to believe at some point the module files were downloaded.

I did a locate for misp-modules and there were a bunch of files that returned from that search.

ran the following and recieved these results:

root@osrs-resource1:/home/osrs# sudo apt-get install python3-dev python3-pip libpq5 libjpeg-dev tesseract-ocr libpoppler-cpp-dev imagemagick virtualenv libopencv-dev zbar-tools libzbar0 libzbar-dev libfuzzy-dev build-essential -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
build-essential is already the newest version (12.4ubuntu1).
libjpeg-dev is already the newest version (8c-2ubuntu8).
libfuzzy-dev is already the newest version (2.14-1).
libzbar-dev is already the newest version (0.10+doc-10.1build2).
libzbar0 is already the newest version (0.10+doc-10.1build2).
tesseract-ocr is already the newest version (4.00~git2288-10f4998a-2).
virtualenv is already the newest version (15.1.0+ds-1.1).
zbar-tools is already the newest version (0.10+doc-10.1build2).
imagemagick is already the newest version (8:6.9.7.4+dfsg-16ubuntu6.8).
libpoppler-cpp-dev is already the newest version (0.62.0-2ubuntu2.10).
libpq5 is already the newest version (10.10-0ubuntu0.18.04.1).
python3-dev is already the newest version (3.6.7-1~18.04).
libopencv-dev is already the newest version (3.2.0+dfsg-4ubuntu0.1).
python3-pip is already the newest version (9.0.1-2.3~ubuntu1.18.04.1).
0 upgraded, 0 newly installed, 0 to remove and 222 not upgraded.
root@osrs-resource1:/home/osrs# sudo -u www-data virtualenv -p python3 /var/www/MISP/venv
Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /var/www/MISP/venv/bin/python3
Traceback (most recent call last):
  File "/usr/bin/virtualenv", line 11, in <module>
    load_entry_point('virtualenv==15.1.0', 'console_scripts', 'virtualenv')()
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 724, in main
    symlink=options.symlink)
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 946, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 1278, in install_python
    shutil.copyfile(executable, py_executable)
  File "/usr/lib/python3.6/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
OSError: [Errno 26] Text file busy: '/var/www/MISP/venv/bin/python3'
root@osrs-resource1:/home/osrs#

Then tried this

root@osrs-resource1:/home/osrs# /var/www/MISP/venv/bin/misp-modules -l 127.0.0.1 -s &
[1] 914
root@osrs-resource1:/home/osrs# WARNING [__init__.py:56 - <module>() ] Unable to load pymisp properly: No module named 'deprecated'
WARNING:pymisp:Unable to load pymisp properly: No module named 'deprecated'
cannot import name 'MISPEvent'
2019-12-06 22:00:10,732 - misp-modules - INFO - Launch MISP modules server from package.
INFO:misp-modules:Launch MISP modules server from package.
2019-12-06 22:00:10,733 - misp-modules - INFO - Helper loaded cache
INFO:misp-modules:Helper loaded cache
2019-12-06 22:00:10,734 - misp-modules - INFO - Unable to load MISP modules from package.
INFO:misp-modules:Unable to load MISP modules from package.

So, long story short - I believe the module files are there, but the steps aren't allowing me to use them.

Jimbo

**Comment by adulau** [comment]
Did you install all the Python requirements in the virtual env?

- First activate the venv `. ./venv/bin/activate` in `/var/www/MISP`
- go to `cd /usr/local/src/misp-modules`
- `pip3 install -I -r REQUIREMENTS`
- then install the misp-module `pip3 install .`
- then start `misp-module`

I tested on a recent VM and it works out-of-the-box

**Comment by open-source-rs** [comment]
I want to say that yes I did install them.
I will try this today Adulau...

Jimbo

**Comment by open-source-rs** [comment]
ok - went back to make sure virtual env was activated 
ran through the steps which did load more packages onto the system
went to start modules

root@osrs-resource1:/usr/local/src/misp-modules# /var/www/MISP/venv/bin/misp-modules -l 127.0.0.1 -s &
[1] 9488
root@osrs-resource1:/usr/local/src/misp-modules# WARNING [__init__.py:56 - <module>() ] Unable to load pymisp properly: No module named 'deprecated'
WARNING:pymisp:Unable to load pymisp properly: No module named 'deprecated'
cannot import name 'MISPEvent'
2019-12-08 10:37:48,156 - misp-modules - INFO - Launch MISP modules server from package.
INFO:misp-modules:Launch MISP modules server from package.
2019-12-08 10:37:48,158 - misp-modules - INFO - Helper loaded cache
INFO:misp-modules:Helper loaded cache
2019-12-08 10:37:48,158 - misp-modules - INFO - Unable to load MISP modules from package.
INFO:misp-modules:Unable to load MISP modules from package.

**Comment by clopmz** [comment]
Same problem here in a Ubuntu 18.04.3 host:

(venv) root@iceland:/usr/local/src/misp-modules# sudo -H -u www-data /var/www/MISP/venv/bin/pip install -I .
Processing /usr/local/src/misp-modules
Processing /var/www/.cache/pip/wheels/84/bf/40/2f6ef700f48401ca40e5e3dd7d0e3c0a90e064897b7fe5fc08/tornado-6.0.3-cp36-cp36m-linux_x86_64.whl
Processing /var/www/.cache/pip/wheels/52/41/b0/bf50409fe2b1d3b79afa3eed71b54b3e30fe5b695db2c7ba2e/psutil-5.6.7-cp36-cp36m-linux_x86_64.whl
Collecting redis>=3
  Using cached https://files.pythonhosted.org/packages/32/ae/28613a62eea0d53d3db3147f8715f90da07667e99baeedf1010eb400f8c0/redis-3.3.11-py2.py3-none-any.whl
Building wheels for collected packages: misp-modules
  Building wheel for misp-modules (setup.py) ... done
  Created wheel for misp-modules: filename=misp_modules-1.0-cp36-none-any.whl size=200256 sha256=01bd935fcd24b112042a7c5c07395d1bd720ca98b3545f179e4e1b15c4ad9f43
  Stored in directory: /var/www/.cache/pip/wheels/84/23/c5/f741153ef3c925d089a7d56c8b45251e71b4a058385cbabec3
Successfully built misp-modules
Installing collected packages: tornado, psutil, redis, misp-modules
Successfully installed misp-modules-1.0 psutil-5.6.7 redis-3.3.11 tornado-6.0.3
(venv) root@iceland:/usr/local/src/misp-modules# sudo -H -u www-data /var/www/MISP/venv/bin/misp-modules -l 127.0.0.1 -s
No module named 'misp_modules.modules.expansion._ransomcoindb'
2019-12-22 10:59:56,385 - misp-modules - INFO - Launch MISP modules server from package.
INFO:misp-modules:Launch MISP modules server from package.
2019-12-22 10:59:56,388 - misp-modules - INFO - Helper loaded cache
INFO:misp-modules:Helper loaded cache
2019-12-22 10:59:56,388 - misp-modules - INFO - Unable to load MISP modules from package.
INFO:misp-modules:Unable to load MISP modules from package.
